administration/live-migration.adoc

```diff
@@ -0,0 +1,138 @@
+# Live Migration
+
+Live migration is a process during which a running Virtual Machine Instance
+moves to another compute node while the guest workload continues to run and
+remain accessable.
+
+## Initiate live migration
+
+Live migration is initiated by posting a VirtualMachineInstanceMigration (VMIM)
+object to the cluster.
+The example below starts a migration process for a virtual machine instance
+`vmi-fedora`
+
+```yaml
+apiVersion: kubevirt.io/v1alpha3
+kind: VirtualMachineInstanceMigration
+metadata:
+  name: migration-job
+spec:
+  vmiName: vmi-fedora
+configuration:
+  completionTimeoutPerGiB: 800
+  progressTimeout: 150
+```
+
+An optional configuration can be provided as part of the `VirtualMachineInstanceMigration` object
+
+### configuration
+
+Depending on the type, live migration process will copy virtual machine memory
+pages and disk blocks To the destination.
+During this process non-locked pages and blocks are being copied and become
+free for the instance to use again. To achieve a successful migration, it is
+assumed that the instance will write to the free pages and blocks (pollute the
+pages) at a lower rate than these are being copied.
+
+#### Completion time
+
+In some cases the virtual machine can write to different memory pages / disk
+blocks at a higher rate than these can be copied, which will prevent the
+migration process from completing in a reasonable amount of time.
+In this case, live migration will be aborted if it is running for a long
+perioud of time. The timeout is calculated base on the side of the VMI, it's
+memory and the ephemeral disks that are needed to be copied. The configurable
+parameter `completionTimeoutPerGiB`, which deafults to 800s is the time for GiB
+of data to wait for the migration to be completed before aborting it.
+A VMI with 8Gib of memory will time out after 6400 seconds.
+
+#### Progress timeout
+
+Live migration will also be aborted when it will be noticed that memory doesn't
+make any progress.
+The time to wait for live migration to make progress in transferring data is
+configurable by `progressTimeout` parameter, which defaults to 150s
+
+## Migration Status Reporting
+
+### Condition and migration method
+
+When starting a virtual machine instance, it has also been calculated whether
+the machine is live migratable. The result is being stored in the VMI
+`Conditions`. The calculation can be based on multiple parameters of the VMI,
```

Comment:
Maybe use the jsonpath `VMI.status.conditions`?

Reply:
done